backend/src/router.go

```diff
@@ -176,6 +177,77 @@ func getTutorsUsername(c *gin.Context) {
 	})
 }
 
+
+// Handler for PUT /tutors/:username. Returns an empty json if sucessful. 
+// Errors if:
+//
+//  - There is no authenticated user (401)
+//  - A server issue prevents parsing the JWT (500)
+//  - Path username is different than logged in user (404)
+//  - The user does not exist (404)
+//  - The user is already a tutor (404)
+//
+// Response Schema: {
+//   200
+// }
+// Error Schema: {
+//   error: String
+// }
+
+func putTutorsUsername(c *gin.Context) {
+	
+	token, err := c.Cookie("jwt")
+	if err != nil {
+		c.JSON(401, gin.H{
+			"error": "Requires an authenticated user.",
+		})
+		return
+	}
+	
+	claims, err := ParseJWT(token)
+	if err != nil {
+		c.JSON(500, gin.H{
+			"error": "Unable to parse JWT: " + err.Error() + ".",
+		})
+		return
+	}
+	
+	username := c.Params.ByName("username")
+	
+	if username != claims.Username {
+		c.JSON(404, gin.H{
+			"error": "User " + username + " not logged in.",
+		})
+		return
+	}
+	
+	var users []User
+	DB.Limit(1).Find(&users, "username = ?", claims.Username)
+	
+	if len(users) != 1 {
+		c.JSON(404, gin.H{
+			"error": "User " + claims.Username + " not found.",
+		})
+		return
+	}
+	
+	var tutors []Tutor
+	DB.Find(&tutors, "User__username = ?", users[0].ID) //error in checking this
+	
+	if len(tutors) > 0 {
+		c.JSON(404, gin.H{
```

Comment:
404 wouldn't be the right status here either. There's a variety of theories on what the best one is, but 400 seems to be the least wrong.

https://stackoverflow.com/questions/3825990/http-response-code-for-post-when-resource-already-exists?noredirect=1&lq=1

Reply:
Decided to go with 409

Reply:
Think the status code change here got mixed up, the one above was changed to 409 but that should be a 500.

Reply:
Fixed